nixos/modules/system/activation/activation-script.nix

```diff
@@ -200,6 +206,11 @@ in
   config = {
     system.activationScripts.stdio = ""; # obsolete
 
+    system.activationScripts.diff = lib.mkIf config.system.showSystemDiffOnRebuild ''
+      if [[ -e /run/current-system ]]; then
+        ${config.nix.package}/bin/nix store diff-closures /run/current-system "$systemConfig"
```

Comment:
Wouldn't this fail if the `nix-command` experimental feature is not enabled? Could be addressed by adding an assertion that the feature needs to be enabled, but it's also a bit questionable whether we want such core functionality to depend on experimental Nix features.

Reply:
can just add `--experimental-features 'nix-command'` to this specific invocation of nix.

Reply:
Not to keen on this. We could call `nix-diff` instead which is a third-party dependency but is stable at least and unlikely to change in any major way.

Reply:
nix-diff recently got some bigger changes. Also I don't think we need to draw in haskell things here.

Reply:
I don't think it's a good idea to use experimental features like this here. Notably `config.nix.package` is used, meaning that users can (and will) override it to more recent versions that might have broken backwards compatibility (which is allowed for experimental features, and in fact a CLI stabilisation is [being discussed](https://github.com/NixOS/rfcs/pull/136)).

I think it might be okay if we _pin_ the Nix package to some version that we know works as we expect. Of course this then requires manual work to update this pinned version every once in a while.

A neat way to think about this: Experimental features have an inherent cost that somebody has to pay, and it shouldn't be the NixOS users (by trying to use a feature but it might be broken), instead it should be developers (e.g. by having to update the Nix pin manually)

Reply:
Alternatively, I don't think using `nix-diff` is a problem at all, Haskell is not problematic for closure size:

```
❯ nix build nixpkgs#nix-diff
❯ nix path-info -Sh ./result
/nix/store/yd078i1jkcps24hh2kxgq1gywq93x3cf-nix-diff-1.0.19	 42.2M
```